### ISSUE TITLE
[Feature]: Allow proxy disabling for REST API requests

### ISSUE THREAD

**Issue by thegoodhen** [appsmithorg/appsmith#36072]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

It seems like all requests (http and https) are going through a proxy. ((https://app.appsmith.com/api/v1/actions/execute))
This can slow the request down up to 100 times in some cases for non-self-hosted appsmith instances.
It is also not clearly mentioned anywhere!!
I suggest adding the option to disable this behavior and make the requests direct. For http requests, we should get a note saying that this is not possible due to mixed-content policies of modern browsers. Thank you!

### Why should this be worked on?

Using proxy like this without clearly stating so is very opaque and leads to a significant slowdown that's non-trivial to trace, especially if the response has tens of thousands of json lines.
A workaround is for the Appsmith user to use the fetch api instead directly.

**Comment by Nikhil-Nandagopal** [comment]
@thegoodhen all requests are executed from the server because doing client side execution and substitution is insecure as all authentication information such as API keys would be exposed to the client browser which would be a security hole.
Could you tell us a bit about your use case that requires an API that returns 10k+ lines of JSON?

**Comment by thegoodhen** [comment]
@Nikhil-Nandagopal Well okay then. That sounds somewhat reasonable, although in some cases no API keys are actually used and in those cases this causes a pointless slowdown. I still do believe that the behavior is a bit too opaque, but I do understand the reason for it. 
At least there is the workaround available where you simply use the fetch API, so I suppose it's not a big deal.

As to my use case, I just noticed the extreme slowdown when loading large amounts of data into my custom chart.

Our server-side aggregation is currently set up so that it doesn't kick in unless we zoom out to a window longer than 2 days. 

I am simultaneously loading 5 different metrics, each getting sent every 3 seconds and when there is no aggregation, that gives me 5 metrics * 20 times a minute * 60 times an hour * 24 hours in a day, so 144000 lines in a day, not even counting the timestamps. 

Now obviously the data rate is pointlessly high, we are stress testing our system. In reality, the rate will be one to two orders of magnitude lower. Additionally, I can change the policies for server-side aggregation, but haven't done so yet as I believe the frontend should still handle this just fine:

The bottleneck is not the server-side lookup (milliseconds) or roundtrip time (tens of milliseconds) or even rendering time (I decimate client-side, processing + rendering is tens of ms), but it's the proxy overhead of about 2500ms and then probably some React callback overhead (?) that I wasn't able to track down yet.

So this is not a dealbraker, we can trivially change the aggregation policies (1 minute aggregation by default?), but I just felt like the slowdown from roughly 25ms roundtrip to 2500ms roundtrip is extremely significant, so I reported it.

Thank you very much!
